migration/idempotent/018-metric-rollup-maintenance.sql

```diff
@@ -0,0 +1,169 @@
+CREATE OR REPLACE PROCEDURE _prom_catalog.execute_caggs_refresh_policy(job_id int, config jsonb)
+AS
+$$
+DECLARE
+    _refresh_interval INTERVAL;
+    _refresh_rollups BOOLEAN := (SELECT prom_api.get_automatic_downsample()::BOOLEAN);
+    r RECORD;
+
+BEGIN
+    SET LOCAL search_path = pg_catalog, pg_temp;
+    _refresh_interval := (config ->> 'refresh_interval')::INTERVAL;
+    IF _refresh_interval IS NULL THEN
+        RAISE EXCEPTION 'refresh_interval cannot be null';
+    END IF;
+
+    FOR r IN
+        SELECT
+            mr.id IS NOT NULL AS rollup_exists,
+            m.table_schema,
+            m.table_name
+        FROM _prom_catalog.metric m
+        LEFT JOIN
+            _prom_catalog.metric_rollup mr ON (m.id = mr.metric_id)
+        WHERE is_view AND view_refresh_interval = _refresh_interval
+    LOOP
+        IF (r.rollup_exists AND NOT _refresh_rollups) THEN
```

Comment:
It still feels like we should incorporate this logic into the query so that we don't bother needlessly looping over potentially many rows.